result/array-bracket-spacing.js

```diff
@@ -0,0 +1,118 @@
+/*eslint-env es6*/
+
+var arr = [ 'foo', 'bar' ];
+var [ x, y ] = z;
+
+var arr = ['foo', 'bar'];
+var [x,y] = z;
```

Comment:
__[development][eslint]__
* Error - A space is required after '[' (array-bracket-spacing)
* Error - 'x' is already defined (no-redeclare)
* Error - 'y' is already defined (no-redeclare)
* Error - A space is required before ']' (array-bracket-spacing)
* Error - 'z' is not defined. (no-undef)